### DISCUSSION TITLE
How to load stories dynamically based on folder files ?

### DISCUSSION THREAD

**Discussion by jeremytenjo** [storybookjs/storybook#12049] [Help]
I used to use the `storiesOf` function and `require.contexts` to dynamically create stories based on files in a folder. 

After upgrading to Storybook V6. I get the error `Cannot add a story when not configuring` .

Is there an alternative way to load stories dynamically?

**Comment by albanlorillard** [comment]
Hello,
I've exactly the same problem. In my context, I have a "special" story that load an svg file in order to display each svg icons inside. But, in order to fetch icon, i've no choice to be asynchrone.

```
(async () => {

  const template = await new TemplateIcons(
    'Icons',
    'List of all the SVG icons.',
    `<svg class="icon" aria-hidden="true"><use [attr.href]="'#time-icon' | svgUseHref"></use></svg>`,
    './assets/images/era-icons.svg'
  ).build();

  storiesOf('Guidelines', module)
    .addDecorator(
      moduleMetadata({
        imports: [SvgUseModule],
      })
    )
    .add('Icons', () => ({
      template: template
    }));

})();
```

So I've seen `store.startConfiguration()` in the breaking change but no documentation how to use it ...

**Reply by shilman** [reply]
We don't support asynchronous stories. We'll add support for async loaders in 6.1

https://github.com/storybookjs/storybook/pull/11012

**Reply by jeremytenjo** [reply]
How would a loader enable the creation of dynamic stories ? Are there any examples in the wild we could reference?
